pandoc_plantuml_filter.py

```diff
@@ -24,7 +24,7 @@ def plantuml(key, value, format_, _):
             caption, typef, keyvals = get_caption(keyvals)
 
             filename = get_filename4code("plantuml", code)
-            filetype = get_extension(format_, "png", html="svg", latex="png")
+            filetype = get_extension(format_, "png", html="svg", latex="latex")
```

Comment:
Why exactly do you want to have that?

Reply:
I *assume* that it effectively includes the diagram as a vector graphic - likely leading to smaller file size, maybe selectable text, and higher fidelity printing - but I didn't actually test it out. In my experimentation it sees that pandoc pdf generation really doesn't like `svg` at all, and likes *some* `eps` format diagrams (about 60% in my case), so I end up falling back to png. Was curious when I saw this PR as I hadn't noticed `latex` as an option to explore.

Perhaps its worth establishing a `PLANTUML_FILETYPE` environment variable so that users can override the `get_extension()` results if/when they want to use more exotic formats than png?

Reply:
Yeah, makes sense.
I'd prefer having a configuration option for the file type instead of changing it for every one to `latex`.  

`png` seems to be better default though because it works for almost every case. If you have file size constraints etc. (which is most likely not a big problem any more these days) you could explicitly use the `latex` format.

Reply:
Thanks for the feedback. indeed a bit ugly with the header addition. The reason i chose the  latex, is the output produces a cleaner diagram filling the whole width. There are some drawbacks though and i have noticed that when the figure goes over a page. the png would be a cleaner solution in this case. I would still like to keep it in. I can add some code to check if the header_includes has the tikz packages, if so export latex for the pdf if not export png.

Reply:
I will withdraw this request and submit 2 new pull requests, one with the split and one with latex plantuml activated via the header_include